src/transforms/remap.rs

```diff
@@ -59,21 +62,23 @@ impl Remap {
 
         Ok(Remap {
             program,
-            drop_on_err: config.drop_on_err,
+            drop_on_error: config.drop_on_error,
+            drop_on_abort: config.drop_on_abort,
         })
     }
 }
 
 impl FunctionTransform for Remap {
     fn transform(&mut self, output: &mut Vec<Event>, mut event: Event) {
-        let original_event = if !self.drop_on_err && self.program.is_fallible() {
-            // We need to clone the original event, since it might be mutated by
-            // the program before it aborts, while we want to return the
-            // unmodified event when an error occurs.
-            Some(event.clone())
-        } else {
-            None
-        };
+        let original_event =
+            if !(self.drop_on_error || self.drop_on_abort) && self.program.is_fallible() {
```

Comment:
I think this isn't quite right. If I have:

```
drop_on_error = false
drop_on_abort = true
```

And my program is fallible, this will result in the original event not being cloned. Such that, if I have an error in the script, it will drop it anyway by virtue of `original_event` being `None`.

I'm thinking maybe this should be:

```
if !(self.drop_on_error && self.drop_on_abort) && self.program.is_fallible()
```

Ideally we might introduce a new `is_abortable()`, separate from `is_falliable()` indicating whether there is an `abort` in the script, that we could use separately with `self.drop_on_abort` to decide whether or not we need to clone. Having `is_fallible` being `true` if `abort` is used feels a bit awkward to me anyway though I can see why you did that to be able to have this conditional here..